devices/GpsDevice/NMEA0183Parser.cs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Iot.Device.Common.GpsDevice
+{
+    /// <summary>
+    /// Provides methods for parsing NMEA0183 data from a GPS device.
+    /// </summary>
+    public static class NMEA0183Parser
+    {
+        /// <summary>
+        /// Parses the GNGLL NMEA0183 data from a GPS device.
+        /// </summary>
+        /// <param name="inputData">The valid GNGLL data string starting with $GNGLL.</param>
+        /// <returns>The parsed GngllData object.</returns>
+        public static GngllData ParaseGngll(string inputData)
```

Comment:
as all the NMEA string are fixed, can you create a kind of generic parser?
Something that can give one of the GngllData or GngsaData file?
I would see a generalized based class names "NmeaData" which contains a field with the signature.
As an example, it could look like:

```csharp
public interface NmeaData
{
    public string Id {get; }
    public NmeaData Parse(string inputData);
}

public class GngsaData: NmeaData
{
    public override string Id = "$GNGSA";
   public NmeaData Parse(string inputData);
    // Rest of the class
}

// And in the parser

public static class NMEA0183Parser
{
    public NmeaParser[] MneaDatas {get; protected set; }

      // In the constructor, just add all the NmeaData derivated class that are supported
     //  MneaDatas = new MneaData[] { GngsaData }; // add all supported

    public NmeaData Parse(string inputData)
    {
        var data = inputData.Split(',');
        for each (string id in NmeaDatas)
        {
            if (id.Id == data[0])
            {
                return id.Parse(inputData);
            }
        }
    }
}
```

Advantage: you can add later on any parser you'd like!